src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -363,16 +363,33 @@ private void updateUninitializedVariables(Deque<DetailAST> prevScopeUnitializedV
     }
 
     /**
-     * If token is LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, or LITERAL_ELSE, then do not
-     * update the uninitialized variables.
+     * If token is LITERAL_IF and there is an {@code else} following or token is CASE_GROUP and
+     * there is another {@code case} following, then update the uninitialized variables.
      * @param ast token to be checked
      * @return true if should be updated, else false
      */
     private static boolean shouldUpdateUninitializedVariables(DetailAST ast) {
-        return ast.getType() != TokenTypes.LITERAL_TRY
-                && ast.getType() != TokenTypes.LITERAL_CATCH
-                && ast.getType() != TokenTypes.LITERAL_FINALLY
```

Comment:
Got it. 

> we should have had some test that at least looked at true/false conditions of that method for `catch`/`finally`

That makes sense. In the original code, the method should return false and it is correct. In the new code it still return false when leaving `catch`/`finally`. So the tests about `catch`/`finally` won't change behaviors. 

> we are missing more regression cases

I agree with you. As we can see in the diff report, the diff are all decreasing of violations (false positives). We are not having wrong `catch`/`finally` samples but missing `method def`/`lambda`/`synchronized` ... samples.

I will add all the false positives from diff report into the input of false positives test.
If it is still missing some specific cases, I will try to write some inputs manually.